File: 238-product-of-array-except-self/product-of-array-except-self.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int length = nums.length;
        int[] left = new int[length];
        int[] right = new int[length];
        int[] result = new int[length];

        left[0] = nums[0];
        right[length - 1] = nums[length - 1];

        for (int i = 1; i < length; i++) {
            left[i] = nums[i] * left[i - 1];
        }
        for (int i = length - 2; i >= 0; i--) {
            right[i] = nums[i] * right[i + 1];
        }
        result[0]=right[1];
        result[length-1]=left[length-2];
        for(int i=1;i<length-1;i++){
            result[i]=left[i-1]*right[i+1];
        }
        return result;

    }
}

File: 238-product-of-array-except-self/product-of-array-except-self.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        left = nums[:]
        right = nums[:]

        for i in range(1, len(nums)):
            left[i] = left[i] * left[i - 1]
        for i in range(len(nums) - 2, -1, -1):
            right[i] = right[i] * right[i + 1]
        nums[0] = right[1]
        nums[len(nums) - 1] = left[len(nums) - 2]

        for i in range(1, len(nums) - 1):
            nums[i] = left[i - 1] * right[i + 1]
        return nums
